ladybug_comfort/cli/utci.py

```diff
@@ -0,0 +1,364 @@
+
+1) Thermo-physiological significance in the whole range of heat
+    exchange conditions of existing thermal environments
+2) Valid in all climates, seasons, and scales
+3) Useful for key applications in human biometeorology.
+
+While UTCI is designed to be valid in all climates and seasons, it assumes
+that human subjects are walking (with a metabolic rate around 2.4 met) and
+that they naturally adapt their clothing with the outdoor temperature.
+For outdoor situations that do not fit these criteria, the Physiological
+Equivalent Temperature (PET) model is recommended.
+"""
+import numpy as np
+
+
+def universal_thermal_climate_index(ta, tr, vel, rh):
+    """Calculate Universal Thermal Climate Index (UTCI) using a polynomial approximation.
+
+    This function here is a Python version of the original UTCI_approx
+    application written in Fortran. Version a 0.002, October 2009
+    The original Fortran code can be found at www.utci.org.
+
+    Note:
+        [1] Peter Bröde, Dusan Fiala, Krzysztof Blazejczyk, Yoram Epstein,
+        Ingvar Holmér, Gerd Jendritzky, Bernhard Kampmann, Mark Richards,
+        Hannu Rintamäki, Avraham Shitzer, George Havenith. 2009.
+        Calculating UTCI Equivalent Temperature. In: JW Castellani & TL
+        Endrusick, eds. Proceedings of the 13th International Conference
+        on Environmental Ergonomics, USARIEM, Natick, MA.
+
+    Args:
+        ta: Air temperature [C]
+        tr: Mean radiant temperature [C]
+        vel: Wind speed 10 m above ground level [m/s].
+            Note that this meteorological speed at 10 m is simply 1.5 times the
+            speed felt at ground in the original Fiala model used to build UTCI.
+        rh: Relative humidity [%]
+
+    Returns:
+        UTCI_approx -- The Universal Thermal Climate Index (UTCI) for the input
+        conditions as approximated by a 4-D polynomial.
+    """
+    # set upper and lower limits of air velocity according to Fiala model scenarios
+    vel = np.where(vel < 0.5, 0.5, np.where(vel > 17, 17, vel))
+
+    # metrics derived from the inputs used in the polynomial equation
+    eh_pa = saturated_vapor_pressure_hpa(ta) * (rh / 100.0).astype(np.float32)  # partial vapor pressure
+
+    pa_pr = eh_pa / 10.0  # convert vapour pressure to kPa
+    d_tr = tr - ta  # difference between radiant and air temperature
+
+    # pre-calculate powers so we can re-use them
+    ta2 = ta ** 2
+    ta3 = ta ** 3
+    ta4 = ta ** 4
+    ta5 = ta ** 5
+    ta6 = ta ** 6
+    vel2 = vel ** 2
+    vel3 = vel ** 3
+    vel4 = vel ** 4
+    vel5 = vel ** 5
+    vel6 = vel ** 6
+    d_tr2 = d_tr ** 2
+    d_tr3 = d_tr ** 3
+    d_tr4 = d_tr ** 4
+    d_tr5 = d_tr ** 5
+    d_tr6 = d_tr ** 6
+    pa_pr2 = pa_pr ** 2
+    pa_pr3 = pa_pr ** 3
+    pa_pr4 = pa_pr ** 4
+    pa_pr5 = pa_pr ** 5
+    pa_pr6 = pa_pr ** 6
+
+    # UTCI approximation calculation
+    utci_approx = ta + \
+        0.607562052 + \
```

Comment:
I would pull out the polynomial into a `_hidden` method that you import from the base `utci.py` module. This way, we don't need to have ~200 lines of code that is copy/pasted unnecessarily.

Reply:
I created a hidden function in the base module.